pkg/analysis/passes/binarypermissions/binarypermissions.go

```diff
@@ -0,0 +1,99 @@
+package binarypermissions
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/grafana/plugin-validator/pkg/analysis"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/archive"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/metadata"
+)
+
+var (
+	binaryExecutableFound       = &analysis.Rule{Name: "binary-executable-found", Severity: analysis.Error}
+	binaryExecutablePermissions = &analysis.Rule{Name: "binary-executable-permissions", Severity: analysis.Error}
+)
+
+var REQUIRED_PERMISSIONS = fs.FileMode.Perm(0755)
+
+var Analyzer = &analysis.Analyzer{
+	Name:     "archivename",
+	Requires: []*analysis.Analyzer{metadata.Analyzer, archive.Analyzer},
+	Run:      run,
+	Rules:    []*analysis.Rule{binaryExecutableFound, binaryExecutablePermissions},
+}
+
+var binarySuffixes = []string{
+	"_linux_amd64",
+	"_linux_arm64",
+	"_darwin_amd64",
+	"_darwin_arm64",
+	"_windows_amd64.exe",
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	archiveDir := pass.ResultOf[archive.Analyzer].(string)
+	metadataBody := pass.ResultOf[metadata.Analyzer].([]byte)
+
+	var data metadata.Metadata
+	if err := json.Unmarshal(metadataBody, &data); err != nil {
+		return nil, err
+	}
+
+	if data.Executable == "" && binaryExecutableFound.ReportAll {
+		binaryExecutableFound.Severity = analysis.OK
+		pass.ReportResult(pass.AnalyzerName, binaryExecutableFound, "No executable defined in plugin.json", "")
+		return nil, nil
+	}
+
+	executable := data.Executable
+	executableParentDir := filepath.Join(archiveDir, filepath.Dir(executable))
+	executableName := filepath.Base(executable)
+
+	var foundBinaries = []string{}
```

Comment:
Nit/tip: this can become

```go
var foundBinaries []string
```

The zero-value for a slice is nil, but in this case it behaves in the same way.

Just personal preference, both are fine :)

Reply:
I like to write explicit code rather than implicit. :)